dawn/src/dawn/CodeGen/Cuda-ico/ASTStencilBody.h

```diff
@@ -39,36 +39,22 @@ class StencilMetaInformation;
 namespace codegen {
 namespace cudaico {
 
-using MergeGroupMap =
-    std::map<int, std::vector<std::vector<std::shared_ptr<ast::ReductionOverNeighborExpr>>>>;
-
 class FindReduceOverNeighborExpr : public ast::ASTVisitorForwardingNonConst {
-  std::vector<std::shared_ptr<ast::ReductionOverNeighborExpr>> foundReductions_;
-  std::map<int, int> reductionToBlock_;
-  int currentBlock_ = -1;
+  bool found_ = false;
 
 public:
   void visit(const std::shared_ptr<ast::ReductionOverNeighborExpr>& expr) override {
-    foundReductions_.push_back(expr);
-    reductionToBlock_.insert({expr->getID(), currentBlock_});
+    found_ = true;
     return;
   }
   void visit(const std::shared_ptr<ast::BlockStmt>& stmt) override {
-    currentBlock_ = stmt->getID();
+    // TODO do we need this override ?
```

Comment:
This looks like an exact code clone of `ASTVISITORFORWARDING_VISIT_IMPL` to me, should be able to safely remove this

Reply:
done